doc/docusaurus/docs/using-plutus-tx/compiling-plutus-tx.md

```diff
@@ -1,95 +1,67 @@
+myPlutusTxCodeCompiled :: CompiledCode (Integer -> Integer)
+myPlutusTxCodeCompiled = $$(compile [|| myPlutusTxCode ||])
 ```
 
-`-fno-ignore-interface-pragmas` and `-fno-omit-interface-pragmas` ensure unfoldings of Plutus Tx functions are available.
-The rest are GHC optimizations that are generally bad for Plutus Tx, and should thus be turned off.
+Under the hood, it uses [`addCorePlugin`](https://hackage.haskell.org/package/template-haskell/docs/Language-Haskell-TH-Syntax.html#v:addCorePlugin) from the `template-haskell` package to install the plugin into the compilation pipeline.
 
-These flags can be specified either in a Haskell module, for example:
-```
-    {-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
-```
-or in a build file.
-For example, if your project is built using Cabal, you can add the flags to the `.cabal` files, like so:
+You can then compile module `B` as you would any regular Haskell module.
+The resulting `CompiledCode` contains the UPLC code, and also includes PIR for debugging.
 
-> ghc-options:
->
-> :   -fno-ignore-interface-pragmas
+Template Haskell is a complicated piece of machinery, but as you can see, you need to understand almost none of it for the purpose of compiling Plutus Tx.
 
-### Plutus Tx compiler options
+This method is preferred since it can leverage Template Haskell's [`location`](https://hackage.haskell.org/package/template-haskell/docs/Language-Haskell-TH.html#v:location) function to pass the location of `$$(compile [|| ... ||])` to the plugin, which is used in error messages.
 
-> :pushpin: **NOTE**
->
-> This section only covers GHC flags, not Plutus Tx compiler flags.
-> A number of options can be passed to the Plutus Tx compiler.
-> See [Reference > Plutus Tx Compiler Options](../delve-deeper/plutus-tx-compiler-options.md) for details.
+## Compiling Using GHC Flag
+
+An alternative way to compile `myPlutusTxCode` is by using the `-fplugin` GHC flag, which installs a plugin into the pipeline.
+Use this flag with the `plc` function:
 
-### Pragmas
+```haskell
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+module B where
 
-All functions and methods should have the `INLINEABLE` pragma, so that their unfoldings are made available to the Plutus Tx compiler.
+import Data.Proxy
+import PlutusTx.Code (CompiledCode)
+import PlutusTx.Plugin (plc)
 
-The `-fexpose-all-unfoldings` flag also makes GHC expose all unfoldings, but unfoldings exposed this way can be more optimized than unfoldings exposed via `INLINEABLE`.
-In general, we do not want GHC to perform optimizations, since GHC optimizes a program based on the assumption that it has non-strict semantics and is evaluated lazily (call by need), which is not true for Plutus Tx programs.
-Therefore, `INLINEABLE` is preferred over `-fexpose-all-unfoldings`, even though the latter is simpler.
+myPlutusTxCodeCompiled :: CompiledCode (Integer -> Integer)
+myPlutusTxCodeCompiled = plc (Proxy @"location info") myPlutusTxCode
```

Comment:
maybe mention that "location info" is for debugging purposes and not very important